### PR TITLE
Run functional tests against sandbox-bundled

### PR DESCRIPTION
**Pull request by jeevb**

### CODE REVIEW

**Comment by jeevb** [comment]
Blocked by: https://github.com/flyteorg/flytesnacks/pull/983

**Review by wild-endeavor — APPROVED** [review]

File: .github/workflows/single-binary.yml [hunk 1]
```diff
@@ -121,25 +118,85 @@ jobs:
             FLYTECONSOLE_VERSION=${{ env.FLYTECONSOLE_VERSION }}
             FLYTE_VERSION=${{ env.FLYTE_VERSION }}
           file: Dockerfile
-          push: ${{ ( github.event_name == 'push' && github.ref == 'refs/heads/master' ) || github.event_name == 'workflow_dispatch' }}
-      - name: Upload Artifact GitHub Action
-        uses: actions/upload-artifact@v2
+          push: true
+
+  sandbox-bundled-functional-tests:
```
[the remark this answers is not in the document]
Comment (jeevb) [reply .github/workflows/single-binary.yml:123]: Yes. This whole pipeline is possible to trigger using `workflow_dispatch`.

File: .github/workflows/single-binary.yml [hunk 2]
```diff
@@ -121,25 +118,85 @@ jobs:
...
+          driver-opts: image=moby/buildkit:master
+          buildkitd-flags: "--allow-insecure-entitlement security.insecure"
+      - name: Build sandbox image for functional tests
+        uses: docker/build-push-action@v3
+        with:
+          context: docker/sandbox-bundled
+          load: true
+          allow: "security.insecure"
+          tags: flyte-sandbox-bundled:local
+      - name: Prune Docker Buildx cache to reclaim storage
```
[the remark this answers is not in the document]
Comment (jeevb) [reply .github/workflows/single-binary.yml:146]: Without this, the GHA runner is under disk pressure and evicts all the pods.

[the remark this answers is not in the document]
Comment (wild-endeavor) [reply .github/workflows/single-binary.yml:146]: can you add as comment?

[the remark this answers is not in the document]
Comment (jeevb) [reply .github/workflows/single-binary.yml:148]: Done.

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**